### PR TITLE
Refactor ListCommand and related code

### PR DESCRIPTION
**Pull request by luodan01**
Refactored code related to the List command. In test cases, renamed all instances of names AMY to AKIRA, BOB to BNHA; tags HUSBAND to SHOUNEN and FRIEND to SUPERHERO

### CODE REVIEW

File: src/test/java/seedu/anilist/logic/commands/EditCommandTest.java [hunk 1]
```diff
@@ -81,12 +81,12 @@ public void execute_noFieldSpecifiedUnfilteredList_success() {
...
+        showAnimeAtIndex(model, INDEX_FIRST_ANIME);
 
         Anime animeInFilteredList = model.getFilteredAnimeList().get(INDEX_FIRST_ANIME.getZeroBased());
-        Anime editedAnime = new AnimeBuilder(animeInFilteredList).withName(VALID_NAME_BOB).build();
+        Anime editedAnime = new AnimeBuilder(animeInFilteredList).withName(VALID_NAME_BNHA).build();
         EditCommand editCommand = new EditCommand(INDEX_FIRST_ANIME,
-                new EditPersonDescriptorBuilder().withName(VALID_NAME_BOB).build());
+                new EditAnimeDescriptorBuilder().withName(VALID_NAME_BNHA).build());
 
         String expectedMessage = String.format(EditCommand.MESSAGE_EDIT_PERSON_SUCCESS, editedAnime);
```
Comment (muhammad-faruq) [inline src/test/java/seedu/anilist/logic/commands/EditCommandTest.java:91]: Is this part cannot change cos its part of edit command?

File: src/test/java/seedu/anilist/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -97,30 +97,30 @@ public void execute_filteredList_success() {
...
 
     @Test
-    public void execute_duplicatePersonUnfilteredList_failure() {
+    public void execute_duplicateAnimeUnfilteredList_failure() {
         Anime firstAnime = model.getFilteredAnimeList().get(INDEX_FIRST_ANIME.getZeroBased());
-        EditCommand.EditAnimeDescriptor descriptor = new EditPersonDescriptorBuilder(firstAnime).build();
+        EditCommand.EditAnimeDescriptor descriptor = new EditAnimeDescriptorBuilder(firstAnime).build();
         EditCommand editCommand = new EditCommand(INDEX_SECOND_ANIME, descriptor);
 
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_PERSON);
```
Comment (muhammad-faruq) [inline src/test/java/seedu/anilist/logic/commands/EditCommandTest.java:105]: similarly there is still person here

File: src/test/java/seedu/anilist/logic/commands/EditCommandTest.java [hunk 3]
```diff
@@ -97,30 +97,30 @@ public void execute_filteredList_success() {
...
+        showAnimeAtIndex(model, INDEX_FIRST_ANIME);
 
-        // edit person in filtered list into a duplicate in anime list
+        // edit anime in filtered list into a duplicate in anime list
         Anime animeInList = model.getAniList().getAnimeList().get(INDEX_SECOND_ANIME.getZeroBased());
         EditCommand editCommand = new EditCommand(INDEX_FIRST_ANIME,
-                new EditPersonDescriptorBuilder(animeInList).build());
+                new EditAnimeDescriptorBuilder(animeInList).build());
 
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_PERSON);
```
Comment (muhammad-faruq) [inline src/test/java/seedu/anilist/logic/commands/EditCommandTest.java:117]: ^

File: src/test/java/seedu/anilist/logic/parser/AnimeListParserTest.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public void parseCommand_delete() throws Exception {
     @Test
     public void parseCommand_edit() throws Exception {
         Anime anime = new AnimeBuilder().build();
-        EditCommand.EditAnimeDescriptor descriptor = new EditPersonDescriptorBuilder(anime).build();
+        EditCommand.EditAnimeDescriptor descriptor = new EditAnimeDescriptorBuilder(anime).build();
         EditCommand command = (EditCommand) parser.parseCommand(EditCommand.COMMAND_WORD + " "
                 + INDEX_FIRST_ANIME.getOneBased() + " " + PersonUtil.getEditPersonDescriptorDetails(descriptor));
```
Comment (muhammad-faruq) [inline src/test/java/seedu/anilist/logic/parser/AnimeListParserTest.java:60]: PersonUtil

File: src/test/java/seedu/anilist/logic/commands/EditCommandTest.java (hunk 1 above)
Comment (luodan01) [reply src/test/java/seedu/anilist/logic/commands/EditCommandTest.java:91]: Ok I will refactor edit command as well, wasn't sure previously if I should do that in this pr

**Pull request merged**